inst/tests/tests.Rraw

```diff
@@ -13626,6 +13626,23 @@ dx = data.table(id = 1L, key = "id")
 di = list(z=c(2L, 1L))
 test(1999.2, key(dx[di]), NULL)
 
+# test the non-ASCII strings are correctly sorted on Windows, #3397
+if (.Platform$OS.type == 'windows') local({
+  lc_collate <- Sys.getlocale(c('LC_COLLATE'))
+  lc_ctype <- Sys.getlocale(c('LC_CTYPE'))
+  Sys.setlocale('LC_COLLATE', "Chinese (Simplified)_China.936")
+  Sys.setlocale('LC_CTYPE', "Chinese (Simplified)_China.936")
+  on.exit({
+    Sys.setlocale('LC_COLLATE', lc_collate)
+    Sys.setlocale('LC_CTYPE', lc_ctype)
+  }, add = TRUE)
```

Comment:
Not sure if `add=TRUE` is appropriate here. Locale setting is properly reverted to old when `local()` finishes. `add=TRUE` make sense if there would be any other preceding `on.exit` calls inside `local`.

Reply:
It's the same with or without `add=TRUE` . It's just a good practice (in  my opinion)  to always add this to avoid overriding the previous `on.exist()` logic.